src/module/skb_drop/event.rs

```diff
@@ -30,18 +30,17 @@ impl EventFmt for SkbDropEvent {
     }
 }
 
-#[derive(Default)]
 #[event_section_factory(SkbDropEvent)]
 pub(crate) struct SkbDropEventFactory {
     /// Map of u32 to skb free reasons. It is filled lazyly to avoid executing
     /// `parse_drop_reasons` on a machine not collecting events. A `Some` value
     /// with an empty map means we couldn't retrieve the drop reasons from the
     /// running kernel.
-    reasons: Option<HashMap<i32, String>>,
+    reasons: HashMap<i32, String>,
 }
 
 impl SkbDropEventFactory {
-    pub(crate) fn parse_drop_reasons(&mut self) -> Result<()> {
```

Comment:
I also faced this issue when writing the ct module.
The problem here is that the all the section factories are also built when we unmarshal from json file so now, in order to simply list the events or sort them, we need to have access to the BTF (with the associated permissions). This is not OK IMHO.

I did not solve this problem elegantly xD
I basically stored a boolean in the `Collector` object that is set to `true` on `Collector::init()`, this indicates that we do need BPF decoding. I then pass that boolean down to the `EventSectionFactory` constructor and only use BTF  if it's `true`.  
 
A more elegant way would require API modification: splitting the factories, passing a FactoryType enum, or something like that